### ISSUE TITLE
Up-speed and down-speed would be nice

### ISSUE THREAD

**Issue by sathishvj** [ddo/fast#1]
I'm assuming what you are showing now is only down-speed.

**Comment by ddo** [comment]
you're right. fast.com only test your download speed i think. i will update the description ty :+1:

**Comment by 161134050** [comment]
it also does upload if you click "Show more info"

**Comment by D34DC3N73R** [comment]
[This fast package](https://github.com/sindresorhus/fast-cli) allows for upload testing, but I'd prefer to keep using this snap package vs node js.

**Comment by quarky42** [comment]
Fast.com does not only test your download speed.  It is just that the upload speed test is "hidden" until you expose it.  It would be very nice if the upload speed test were added to this.

In a browser, go to fast.com.   Wait for the test to complete.   Below the test results there will be a new button appear to "Show More Info".  When you click that button the upload speed test will be performed.  There are options for controlling the number of simultaneous streams (which is good for getting better results at the cost of taking more bandwidth on your network temporarily) as well as an option for minimum speed test length in terms of time.  The time setting is VERY handy if you want to watch the test results over a long period of time looking for issues where the bandwidth drops.  When using wireless internet providers there can be interference that causes your connection to drop severely, just for one example.

Having access to all of the settings that are available on a Fast.com speedtest would be VERY helpful.  Please consider adding these.  

The first screenshot is of a basic test being performed for the first time:
![Fast com_SpeedTest_BasicInitialTest](https://user-images.githubusercontent.com/48300108/107885884-f0a0b080-6eb9-11eb-8038-bfe80b992573.jpg)

At the end, when you click the "Show more info" button it displays a screen like this, but I have already entered in my preferred settings.  The default minimum stream is 1 to 8.  I prefer 4 to 8 and get better / more consistent results this way.  I also mark the "Always show all metrics" setting because I do in fact want to see the upload speed and I mark the option to save these settings for future use "Save config for this device":
![Fast com_SpeedTest_Settings](https://user-images.githubusercontent.com/48300108/107885903-10d06f80-6eba-11eb-938c-4758b8ed1ee2.jpg)

Which now when I do a speed test gives me these results:
![Fast com_SpeedTest_WithAlwaysShowAllMetrics](https://user-images.githubusercontent.com/48300108/107885941-5725ce80-6eba-11eb-824c-59d796d353c6.jpg)

So, yes please, add upload testing to this package.  It would be very useful, especially if the user can control all those same settings.  Here is the contents of the cookie that gets saved when you enable that option.:
![Fast com_SpeedTest_LocalCookieValues](https://user-images.githubusercontent.com/48300108/107886185-85f07480-6ebb-11eb-8cca-03951bcfb04f.jpg)

Parallel connections  (minConnections, maxConnections) has a range of 1 to 30.   (The field goes up to 64, but it won't let you save it.  30 is the maximum.)
Duration (minDuration, maxDuration) has a range of 1 to 300.

shouldPersist is the boolean that saves settings each time.
showAdvanced is the boolean that displays the upload speed and other measurements.

I hope this helps.

**Comment by parsley72** [comment]
Upload is available: https://github.com/sindresorhus/fast-cli/commit/076a30d178bbfde3972f5adfc597adfbd26314f3
